rstudio-tidyverse/rockcraft.yaml

```
@@ -0,0 +1,261 @@
+      gpg --keyserver keys.gnupg.net --keyserver pgp.surfnet.nl --recv-keys 3F32EE77E331692F 
+      gpg --keyserver keys.openpgp.org --recv-keys 51C0B5BB19F92D60 
+      dpkg-sig --verify /tmp/rstudio-server.deb 
+      rm -rf "${GNUPGHOME}" 
+      
+      # install rstudio-server
+      dpkg -i /tmp/rstudio-server.deb 
+      rm -f /tmp/rstudio-server.deb 
+                      
+      # Install - r-tidyverse                  
+      "${CONDA_BIN}/conda" install -y -q \
+        r-tidyverse="${R_TIDYVERSE_VERSION}" 
+                      
+      # Clean                  
+      "${CONDA_BIN}/conda" clean -a -f -y      
+      
+      # Create some directories for staging files
+      mkdir -p "${CRAFT_PART_INSTALL}/${CONDA_DIR}" \
+      "${CRAFT_PART_INSTALL}/etc" \
+      "${CRAFT_PART_INSTALL}/run" \
+      "${CRAFT_PART_INSTALL}/${HOME}" \
+      "${CRAFT_PART_INSTALL}/${HOME_TMP}" 
+
+      # Write some config files for activating conda in user sessions
+      chmod 2775 "${CRAFT_PART_INSTALL}/${CONDA_DIR}"
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${HOME}/.bashrc"
+      echo ". /opt/conda/etc/profile.d/conda.sh" >> "${CRAFT_PART_INSTALL}/etc/profile"
+      echo "conda activate base" >> "${HOME}/.bashrc"
+      
+      # Do some cleanup
+      rm -rf "${HOME}/.cache"
+
+      # Stage the directories
+      cp -r "${HOME}" "${CRAFT_PART_INSTALL}/home"
+      cp -r "${CONDA_DIR}" "${CRAFT_PART_INSTALL}/opt"
+      cp -r /etc/rstudio "${CRAFT_PART_INSTALL}/etc/rstudio"
+      cp -r /run/rstudio-server  "${CRAFT_PART_INSTALL}/run/rstudio-server" 
+      cp -r /usr/lib/rstudio-server  "${CRAFT_PART_INSTALL}/usr/lib/rstudio-server"
+      cp -r /var/lib/rstudio-server  "${CRAFT_PART_INSTALL}/var/lib/rstudio-server" 
+      cp "${R_HOME}/etc/Renviron.site" "${CRAFT_PART_INSTALL}/${R_HOME}/etc/Renviron.site"
+      cp "$CRAFT_PART_SRC/components/example-notebook-servers/rstudio/rsession.sh" "${CRAFT_PART_INSTALL}/opt"
+      
+      # Writing config files for rstudio
+      # the default DB path '/var/lib/rstudio-server' causes permission issues when running as a random UID (like in OpenShift)
+      echo "provider=sqlite" > "${CRAFT_PART_INSTALL}/etc/rstudio/database.conf"
+      echo "directory=/tmp/rstudio_db" >> "${CRAFT_PART_INSTALL}/etc/rstudio/database.conf"               
+      # use advisory file-locks to improve PVC support
+      echo "lock-type=advisory" > "${CRAFT_PART_INSTALL}/etc/rstudio/file-locks"             
+      # allow kubeflow to display rstudio in an iframe
+      echo "www-frame-origin=same" >> "${CRAFT_PART_INSTALL}/etc/rstudio/rserver.conf" 
+
+  non-root-user:
+    plugin: nil
+    after: [rstudio-tidyverse]
+    build-environment:
+      - NB_USER: "_daemon_"
+      - HOME: "/home/jovyan"
+      - HOME_TMP: "/tmp_home/jovyan"
+      - NB_UID: "1000"
+      - NB_GID: "0"
+      - USERS_GID: "100"
+      - CONDA_DIR: "/opt/conda"
+      - R_HOME: "${CONDA_DIR}/lib/R"
```

suggestion: We don't use some environment variables at all in this part, so we can safely delete this
```suggestion
    build-environment:
      - HOME: "/home/jovyan"
      - HOME_TMP: "/tmp_home/jovyan"
      - NB_GID: "0"
      - CONDA_DIR: "/opt/conda"
      - R_HOME: "${CONDA_DIR}/lib/R"
```